### ISSUE TITLE
Build seems to succeed but installation fails in Amazon Linux 2 Docker image

### ISSUE THREAD

**Issue by mbklein** [asdf-vm/asdf-erlang#249]
### Description

I'm trying to use `asdf` to install Erlang in an Amazon Linux 2 Docker container to be used as a continuous integration base image. The build is written in a `Dockerfile`, but this issue is written with manual, interactive build steps to make it easily reproducible.

### Steps to Reproduce Interactively

[on host system]
```
$ docker run -ti amazonlinux:2
```
[inside docker container]
```
[root /] # amazon-linux-extras install epel -y 
[root /] # yum groupinstall -y -d1 'Development Tools' 'C Development Tools and Libraries'
[root /] # yum install -y -d1 fop java-17-amazon-corretto-devel libxslt ncurses-devel openssl-devel util-linux-user wxGTK3-devel wxBase3 libiodbc unixODBC-devel.x86_64 erlang-odbc.x86_64 
[root /] # useradd -m ci
[root /] # su -l - ci

[ci ~] $ git clone https://github.com/asdf-vm/asdf.git ~/.asdf --branch v0.10.0
[ci ~] $ echo '. ~/.asdf/asdf.sh' >> ~/.bash_profile
[ci ~] $ source ~/.asdf/asdf.sh
[ci ~] $ asdf plugin add erlang
[ci ~] $ asdf install erlang 24.2.1
```

### Expected Behavior

Something that ends with
```
Erlang/OTP 24.2.1 (asdf_24.2.1) has been successfully built
Cleaning up compilation products for
Cleaned up compilation products for  under /home/ci/.asdf/plugins/erlang/kerl-home/builds

[ci ~] $ asdf list erlang
  24.2.1
```

### Actual Behavior

```
[ci ~]$ asdf install erlang 24.2.1
asdf_24.2.1 is not a kerl-managed Erlang/OTP installation
No build named asdf_24.2.1
Downloading 24.2.1 to /home/ci/.asdf/downloads/erlang/24.2.1...
Extracting source code
Building Erlang/OTP 24.2.1 (asdf_24.2.1), please wait...
APPLICATIONS INFORMATION (See: /home/ci/.asdf/plugins/erlang/kerl-home/builds/asdf_24.2.1/otp_build_24.2.1.log)
 * wx             : wxWidgets was not compiled with --enable-webview or wxWebView developer package is not installed, wxWebView will NOT be available
 *         wxWidgets must be installed on your system.
 *         Please check that wx-config is in path, the directory
 *         where wxWidgets libraries are installed (returned by
 *         'wx-config --libs' or 'wx-config --static --libs' command)
 *         is in LD_LIBRARY_PATH or equivalent variable and
 *         wxWidgets version is 3.0.2 or above.

Erlang/OTP 24.2.1 (asdf_24.2.1) has been successfully built
[ci ~]$ echo $?
1
[ci ~]$ asdf list erlang
  No versions installed
```

(Note that the “...has been successfully built” message is _not_ followed by a message about installing or cleaning up)

### Version Details

```
$ uname -a
Linux 283719858e11 5.10.104-linuxkit #1 SMP Wed Mar 9 19:05:23 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
$ asdf --version
v0.10.0-77fd510
```

**Comment by mbklein** [comment]
FWIW, I don't really _need_ jinterface, odbc, wxWidgets, or documentation support. I just installed those dependencies to rule them out as the cause of the failure.

**Comment by silviogreuel** [comment]
Have the same issue on archlinux with the version 24.3.4  (the first version that I've tried to install):

![image](https://user-images.githubusercontent.com/3980096/169902375-b0757e47-e817-4bf1-a518-6e7d5236d938.png)

```
~/workspaces ᐅ asdf uninstall erlang 24.3.4          
No such version
~/workspaces ᐅ asdf install erlang 24.3.4 —-with-odbc 
asdf_24.3.4 is not a kerl-managed Erlang/OTP installation
The asdf_24.3.4 build has been deleted
Downloading 24.3.4 to /home/silvio/.asdf/downloads/erlang/24.3.4...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  102M  100  102M    0     0  77.2M      0  0:00:01  0:00:01 --:--:-- 98.9M
Extracting source code
Building Erlang/OTP 24.3.4 (asdf_24.3.4), please wait...
APPLICATIONS DISABLED (See: /home/silvio/.asdf/plugins/erlang/kerl-home/builds/asdf_24.3.4/otp_build_24.3.4.log)
 * jinterface     : No Java compiler found

APPLICATIONS INFORMATION (See: /home/silvio/.asdf/plugins/erlang/kerl-home/builds/asdf_24.3.4/otp_build_24.3.4.log)
 * wx             : wxWidgets was not compiled with --enable-webview or wxWebView developer package is not installed, wxWebView will NOT be available

Erlang/OTP 24.3.4 (asdf_24.3.4) has been successfully built
~/workspaces ᐅ asdf list                             
direnv
  No versions installed
elixir
  No versions installed
erlang
  24.3.3
  25.0
nodejs
  No versions installed
~/workspaces ᐅ uname -a
Linux sloth 5.17.9-arch1-1 #1 SMP PREEMPT Wed, 18 May 2022 17:30:11 +0000 x86_64 GNU/Linux
~/workspaces ᐅ asdf --version
v0.9.0
```

**Comment by silviogreuel** [comment]
My issue was fixed after upgrading asdf to v0.10.1

**Comment by mbklein** [comment]
This seems to have been fixed as of asdf v0.10.1 so I'm going to close this issue out.
